src/uct/ib/rdmacm/rdmacm_cm_ep.c

```diff
@@ -199,41 +180,36 @@ static ucs_status_t uct_rdmacm_cm_create_dummy_cq_qp(struct rdma_cm_id *id,
     qp = ibv_create_qp(id->pd, &qp_init_attr);
     if (qp == NULL) {
         ucs_error("failed to create a dummy ud qp. %m");
-        status = UCS_ERR_IO_ERROR;
-        goto err_destroy_cq;
+        return UCS_ERR_IO_ERROR;
     }
 
     ucs_debug("created ud QP %p with qp_num: 0x%x and cq %p on rdmacm_id %p",
               qp, qp->qp_num, cq, id);
 
-    *cq_p = cq;
     *qp_p = qp;
-
     return UCS_OK;
-
-err_destroy_cq:
-    ibv_destroy_cq(cq);
-err:
-    return status;
 }
 
 ucs_status_t
 uct_rdamcm_cm_ep_set_qp_num(struct rdma_conn_param *conn_param,
                             uct_rdmacm_cm_ep_t *cep)
 {
-    ucs_status_t status;
-    struct ibv_qp *qp;
     struct ibv_cq *cq;
+    ucs_status_t status;
+
+    status = uct_rdmacm_cm_get_cq(uct_rdmacm_cm_ep_get_cm(cep), cep->id->verbs,
+                                  cep->id->pd->handle, &cq);
+    if (status != UCS_OK) {
+        return status;
+    }
 
     /* create a dummy qp in order to get a unique qp_num to provide to librdmacm */
-    status = uct_rdmacm_cm_create_dummy_cq_qp(cep->id, &cq, &qp);
+    status = uct_rdmacm_cm_create_dummy_qp(cep->id, cq, &cep->qp);
```

Comment:
maybe call uct_rdmacm_cm_get_cq from uct_rdmacm_cm_create_dummy_qp()?

Reply:
I would like to avoid hidden side effect of such function, let's keep things simpler

Reply:
ok